sycl/test/esimd/slm_init_noinline_check.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: not %clangxx -fsycl  %s 2>&1 | FileCheck  %s
+
+// This test verifies call to slm_init from a function not marked as
+// always_inline triggers an error.
```

Comment:
nit: the comment is not precise. It tests the case when a function is marked `noinline`, not `not marked as always_inline`.